hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/services/CountryAuthnResponseTranslatorService.java

```diff
@@ -101,11 +127,14 @@ private InboundResponseFromCountry toModel(ValidatedResponse response, Optional<
         CountryAuthenticationStatus status = countryAuthenticationStatusUnmarshaller.fromSaml(response.getStatus());
 
         return new InboundResponseFromCountry(
-            response.getIssuer().getValue(),
-            validatedIdentityAssertionOptional.map(Assertion::getSubject).map(Subject::getNameID).map(NameID::getValue),
-            Optional.ofNullable(status).map(CountryAuthenticationStatus::getStatusCode).map(CountryAuthenticationStatus.Status::name),
-            status.getMessage(),
-            passthroughAssertion.map(assertion -> assertionBlobEncrypter.encryptAssertionBlob(matchingServiceEntityId, assertion.getUnderlyingAssertionBlob())),
-            levelOfAssurance);
+                response.getIssuer().getValue(),
+                validatedIdentityAssertionOptional.map(Assertion::getSubject).map(Subject::getNameID).map(NameID::getValue),
```

Comment:
I know this isn't new, but is there a nicer way of writing this without all the maps?

Reply:
Well, as it is, I'm not entirely sure there is (barring methods to hide what are effectively a bunch of null checks) but it does open the wider question of should Optional<>s be used for anything but return values.